### PR TITLE
Implementation of new mini-game Birdie Dash 

### PR DESCRIPTION
**Pull request by EllaBerglas**
Implementation of mini-game birdie dash from the mini-games menu.
Bird should move with W, up arrow and space and should be able to play the game
Screen should resize, backspace should exit and escape should bring up pause menu.
Score board should be present and increment score according to the coins collected
Game will terminate if it touches the ground or the spikes
End game screen should then boot up
Featur ticket:
#6
Should close task tickets:
#250
#249
#248
#247
#246
#202
#196
#272
#167
#273
Need this pull request done so we can implement birdie dash into main game through the magpie npc so please be nice :)

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/components/minigames/birdieDash/entities/Bird.java [hunk 1]
```diff
@@ -1,4 +1,159 @@
...
+
+/**
+ * Class for the bird in the mini-game birdie dash
+ */
 public class Bird {
-    // To be implemented
-}
+    private final Vector2 position;
+    private final Vector2 velocity;
+    private final Rectangle boundingBox;
```
Comment (reemmi) [inline source/core/src/main/com/csse3200/game/components/minigames/birdieDash/entities/Bird.java:13]: Needs javadocs for all variables

File: source/core/src/main/com/csse3200/game/components/minigames/birdieDash/entities/Bird.java [hunk 2]
```diff
@@ -1,4 +1,159 @@
...
+        isFlapping = false;
+    }
+
+    /**
+     * Updates the birds position
+     * @param deltaTime the time from the last position update
+     * @param multiplier Used to speed up the game
+     */
+    public void update(float deltaTime, float multiplier) {
+
```
Comment (reemmi) [inline source/core/src/main/com/csse3200/game/components/minigames/birdieDash/entities/Bird.java:39]: quite long, consider breaking it up into smaller methods

File: source/core/src/main/com/csse3200/game/components/minigames/birdieDash/rendering/PipeRenderer.java [hunk 3]
```diff
@@ -1,4 +1,72 @@
...
+    private final MinigameRenderer renderer;
+    public PipeRenderer(List<Pipe> pipes, MinigameRenderer renderer) {
+        this.pipes = pipes;
+        this.renderer = renderer;
+        this.renderer.addRenderable(this);
+        loadAssets();
+    }
+
+    /**
+     * Render pipes
```
Comment (reemmi) [inline source/core/src/main/com/csse3200/game/components/minigames/birdieDash/rendering/PipeRenderer.java:29]: add more description to the javadocs this doesn't explain  much

File: source/core/src/main/com/csse3200/game/components/minigames/birdieDash/BirdieDashGame.java [hunk 4]
```diff
@@ -1,3 +1,166 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class for the birdie dash game mechanics
+ */
 public class BirdieDashGame {
+
+    private final int GAME_WIDTH = 1920;
```
Comment (reemmi) [inline source/core/src/main/com/csse3200/game/components/minigames/birdieDash/BirdieDashGame.java:21]: remove variables that are never used

File: source/core/src/main/com/csse3200/game/components/minigames/birdieDash/BirdieDashGame.java [hunk 5]
```diff
@@ -1,3 +1,166 @@
...
+
+    /**
+     * Private method to update all positions of entities in the game
+     * @param dt the change in game time
+     */
+    private void updateGamePosition(float dt) {
+        speedMultiplier += accelerationRate * dt;
+        collisionHandler.checkCollisions();
+        if (collisionHandler.checkSpikes() || bird.touchingFloor()) {
+            isGameOver = true;
```
Comment (reemmi) [inline source/core/src/main/com/csse3200/game/components/minigames/birdieDash/BirdieDashGame.java:120]: just a suggestion :) maybe handle game over in a separate method so this just updates game positions only

File: source/core/src/main/com/csse3200/game/components/minigames/birdieDash/collision/CollisionHandler.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
+package com.csse3200.game.components.minigames.birdieDash.collision;
+
+import com.csse3200.game.components.minigames.birdieDash.entities.Bird;
+import com.csse3200.game.components.minigames.birdieDash.entities.Coin;
+import com.csse3200.game.components.minigames.birdieDash.entities.Pipe;
+import com.csse3200.game.components.minigames.birdieDash.entities.Spike;
+
+import java.util.List;
+
```
Comment (reemmi) [inline source/core/src/main/com/csse3200/game/components/minigames/birdieDash/collision/CollisionHandler.java:9]: this class needs documentation

**Review by reemmi — APPROVED** [review]
Game looks good :) Just need to add javadocs throughout all classes or add descriptive javadocs to properly help explain what each method is doing and break up longer methods to help with readibility specifically in the screens.

File: source/core/src/main/com/csse3200/game/screens/EndMiniGameScreen.java [hunk 7]
```diff
@@ -225,7 +233,14 @@ private void renderEndMessage() {
             @Override
             public void clicked(InputEvent event, float x, float y) {
                 dispose();
-                game.setScreen(new SnakeScreen(game, oldScreen, oldScreenServices));
+                if (gameName == SNAKE) {
```
[the remark this answers is not in the document]
Comment (EllaBerglas) [reply source/core/src/main/com/csse3200/game/screens/EndMiniGameScreen.java:236]: Will do, will take a fair bit of mucking around at the moment but will make a priority for sprint 3.

File: source/core/src/main/com/csse3200/game/ui/minigames/ScoreBoard.java [hunk 8]
```diff
@@ -7,67 +7,97 @@
...
      */
-    public SnakeScoreBoard(int initialScore) {
+    public ScoreBoard(int initialScore, MiniGameNames gameName) {
+        this.initialScore = initialScore;
+        this.gameName = gameName;
+
+        scoreBoardSetUp();
+    }
+
+    private void scoreBoardSetUp () {
```
[the remark this answers is not in the document]
Comment (EllaBerglas) [reply source/core/src/main/com/csse3200/game/ui/minigames/ScoreBoard.java:45]: Will do, will take a fair bit of mucking around at the moment but will make a priority for sprint 3.

**Review by lhick2108 — APPROVED** [review]
Looks good, if what's been discussed happens, everything great. Hopefully should connect with the minigame lootboxes pretty easy.

**Pull request merged**